src/gentropy/common/spark_helpers.py

```diff
@@ -525,7 +542,7 @@ def get_value_from_row(row: Row, column: str) -> Any:
 
 
 def enforce_schema(
-    expected_schema: t.StructType,
+    expected_schema: t.ArrayType | t.StructType | Column | str,
```

Comment:
Are you sure this function can deal with column and string types? This is where the schema is used:
```python
return f.from_json(f.to_json(function(*args, **kwargs)), expected_schema)
```

Essentially we are taking the return value of a function, cast it to json string, then reading it back from json following the schema provided.

Reply:
`to_json` can have any of these parameters - see https://downloads.apache.org/spark/docs/3.3.4/api/python/reference/pyspark.sql/api/pyspark.sql.functions.from_json.html?highlight=from_json